app/desktop/src/main/kotlin/window/WindowsWindowFrame.kt

```diff
@@ -0,0 +1,619 @@
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.FrameWindowScope
+import androidx.compose.ui.window.WindowPlacement
+import androidx.compose.ui.window.WindowState
+import androidx.compose.ui.zIndex
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import me.him188.ani.app.platform.PlatformWindow
+import me.him188.ani.app.platform.window.LayoutHitTestOwner
+import me.him188.ani.app.platform.window.LocalTitleBarThemeController
+import me.him188.ani.app.platform.window.TitleBarThemeController
+import me.him188.ani.app.platform.window.WindowsWindowUtils
+import me.him188.ani.app.platform.window.rememberLayoutHitTestOwner
+import me.him188.ani.app.ui.foundation.ifThen
+import me.him188.ani.app.ui.foundation.layout.LocalCaptionButtonInsets
+import me.him188.ani.app.ui.foundation.layout.LocalPlatformWindow
+import me.him188.ani.app.ui.foundation.layout.LocalTitleBarInsets
+import me.him188.ani.app.ui.foundation.layout.ZeroInsets
+import me.him188.ani.app.ui.foundation.layout.isSystemInFullscreen
+import me.him188.ani.desktop.generated.resources.Res
+import me.him188.ani.desktop.generated.resources.ic_fluent_arrow_minimize_28_regular
+import me.him188.ani.desktop.generated.resources.ic_fluent_dismiss_48_regular
+import me.him188.ani.desktop.generated.resources.ic_fluent_square_48_regular
+import me.him188.ani.desktop.generated.resources.ic_fluent_square_multiple_48_regular
+import me.him188.ani.desktop.generated.resources.ic_fluent_subtract_48_filled
+import org.jetbrains.compose.resources.DrawableResource
+import org.jetbrains.compose.resources.painterResource
+import kotlin.time.Duration.Companion.seconds
+
+@OptIn(ExperimentalTextApi::class, InternalComposeUiApi::class)
+@Composable
+internal fun FrameWindowScope.WindowsWindowFrame(
+    windowState: WindowState,
+    frameState: WindowsWindowFrameState? = rememberWindowsWindowFrameState(),
+    onCloseRequest: () -> Unit,
+    content: @Composable () -> Unit
+) {
+    if (frameState == null) {
+        content()
+        return
+    }
+    val windowUtils = WindowsWindowUtils.instance
+
+    Box(modifier = Modifier.fillMaxSize()) {
+        val isFullScreen = isSystemInFullscreen()
+        //Control the visibility of the title bar. initial value is !isFullScreen.
+        LaunchedEffect(isFullScreen) {
+            frameState.isTitleBarVisible = !isFullScreen
+        }
+
+        // Window content
+        CompositionLocalProvider(
+            LocalTitleBarInsets provides frameState.titleBarInsets,
+            LocalCaptionButtonInsets provides frameState.captionButtonsInsets,
+            LocalTitleBarThemeController provides frameState.titleBarThemeController,
+            content = content,
+        )
+
+        //Hide title bar if window is full screen mode and title bar is not hovered.
```

Comment:
```suggestion
// Hide title bar if window is full screen mode and title bar is not hovered.
```
nit: 注释 slash 和文本之间要有空格